File: tools/playground/testdata/ping/ping.js
var veyron = require('veyron');

veyron.init(function(err, rt){
  if (err) throw err;

  rt.bindTo('pingpong', function(err, s){
    if (err) throw err;

    s.ping('PING', function(err, pong){
      if (err) throw err;

      console.log(pong);
      process.exit(0);
    });
  });
});

File: tools/playground/testdata/src/ping/ping.go
// +build OMIT
package main

import (
	"fmt"

	_ "veyron.io/veyron/veyron/profiles"
	"veyron.io/veyron/veyron2/rt"

	"pingpong"
)

func main() {
	runtime := rt.Init()
	log := runtime.Logger()

	s, err := pingpong.BindPingPong("pingpong")
	if err != nil {
		log.Fatal("error binding to server: ", err)
	}

	pong, err := s.Ping(runtime.NewContext(), "PING")
	if err != nil {
		log.Fatal("error pinging: ", err)
	}
	fmt.Println(pong)
}
